plenum/test/replica/test_instance_faulty_processor.py

```diff
@@ -0,0 +1,166 @@
+from contextlib import ExitStack
+import pytest
+
+from plenum.common.messages.node_messages import BackupInstanceFaulty
+from plenum.server.suspicion_codes import Suspicions
+from plenum.test.replica.helper import check_replica_removed
+from stp_core.loop.eventually import eventually
+from plenum.test.helper import waitForViewChange, create_new_test_node
+from plenum.test.test_node import ensureElectionsDone, checkNodesConnected
+
+
+@pytest.fixture(scope="module", params=[{"instance_degraded": "local", "instance_primary_disconnected": "local"},
```

Comment:
I think we can write unit tests for `BackupInstanceFaultyProcessor`. Please have a look at `fake_node` fixture we can use for it.
I can imagine the following unit tests:
- test that 1 call of `on_backup_degradation` leads to replica removal with local strategy
- test that 1 call of `on_backup_degradation` does not lead to replica removal with quorum strategy
- test that 1 call of `on_backup_primary_disconnected` leads to replica removal with local strategy
- test that 1 call of `on_backup_primary_disconnected` does not lead to replica removal with quorum strategy
- test that `process_backup_instance_faulty_msg` does not lead to removal in case of local strategy
- test that f calls of `process_backup_instance_faulty_msg` with same messages does not lead to removal in case of quorum strategy
- test that f+1 calls of `process_backup_instance_faulty_msg` with same messages leads to removal in case of quorum strategy
- test that BackupInstanceFaulty from our Node is not required to remove replica (just a quorum of any nodes)
- test that only BackupInstanceFaulty with the same viewNo as the current one leads to removal
- test that n BackupInstanceFaulty with different instances where we have f count for every instance in total doesn't lead to removal (in case of quorum)
- test that n BackupInstanceFaulty with different instances where we have f+1 count for 1 instance and f for others lead to removal this 1 instance only
- test that n BackupInstanceFaulty with different instances where we have f+1 for all instances in total leads to removal of all instances

Reply:
1. Added test_on_backup_degradation_for_one with local strategy
2. Added test_on_backup_degradation_for_one with quorum strategy
3. Added test_on_backup_primary_disconnected_for_one with local strategy
4. Added test_on_backup_primary_disconnected_for_one with quorum strategy
5. Was in method __process_backup_instance_faulty_msg_work_with_different_msgs()
6. Added check in __process_backup_instance_faulty_msg_work_with_different_msgs() but I don't sure that this check is correct. Should we sleep some time and what time?
7. In test_process_backup_instance_faulty_msg() we send n-1 message and it's more then f. Is it enaugh?
8. Changed sending in __process_backup_instance_faulty_msg_work_with_different_msgs()
9. Added test_process_backup_instance_with_incorrect_view_no
10. Why shoulden't remove?
11.
12.
13. Was in test_restore_replicas
14. Added test_process_backup_instance_empty_msg